src/unix/haiku.c

```diff
@@ -26,6 +26,7 @@
 #include <FindDirectory.h> /* find_path() */
 #include <OS.h>
 
+#include <private/shared/cpu_type.h>
```

Comment:
This is a private header and I'd prefer that we don't use this, as it could change at any time. Haiku should provide a public interface for this.

Also this would only work for x86 cpus.

Reply:
Currently there is no other public C API that returns the CPU model into a readable string. I used the private API as a last resort to fix a crash whenever the 'model' field was queried.

For now I am only concerned about supporting x86 and x86_64 and this API is very useful in CPU model identification. Other architectures will return "unknown" instead.

As for providing a public API, this will require a header change in 4 applications: AboutHaiku, Pulse, Debugger and sysinfo which shouldn't be too hard to do if this API was made public. But we'll have to see what the other core devs think about this.

Reply:
I think for now I'll hardcode "unknown" or an empty string as the model. We should file a ticket on Trac for getting a proper public API for this.